src/plot_api/subroutines.js

```diff
@@ -125,7 +125,8 @@ function lsInner(gd) {
     // activate mode just for large splom (which benefit the most from this
     // optimization), but this could apply to all cartesian subplots.
     var noNeedForBg = (
-        fullLayout._hasOnlyLargeSploms &&
+        Color.opacity(fullLayout.paper_bgcolor) === 1 &&
+        Color.opacity(fullLayout.plot_bgcolor) === 1 &&
```

Comment:
you don't have to test both, do you? you're just going to check that they match below anyway.

But also if `Color.opacity(fullLayout.plot_bgcolor) === 0` regardless of `paper_bgcolor` - if the user has explicitly said "make the plot transparent" we don't need these rects even if there *is* overlap.

Reply:
> you don't have to test both, do you?

Ha I knew that you were going to make that comment! I left it that way because I thought it made things nicely verbose and self-documented w/o sacrificing perf.

> But also if `Color.opacity(fullLayout.plot_bgcolor) === 0` regardless of `paper_bgcolor` - if the user has explicitly said "make the plot transparent" we don't need these rects even if there _is_ overlap.

Good point. An even better improvement would be to make one `<path d="">` to draw all bg rectangles as a single node. Ok if I leave this for latter?

Reply:
OK, to my mind it's actually more confusing to read this way but I'm not going to insist. Sure, feel free to leave out the transparent case and possible node combination as long as they're recorded in an issue somewhere.